### ISSUE TITLE
Local hashrate fluctuates too much

### ISSUE THREAD

**Issue by newtoopensuse** [ethereum-mining/ethminer#1103]
Hi,
Running 0.14-rc10, win 10 4G of RAM, 6 x rx 570 4G, core 1070, mem 2020, voltage -170 (hwinfo  shows no errors). The local hashrate ranges from 170 to 180. If I run claymore, the rate stays around 176. 

1. Does this have any bad effects on the GPUs as they constantly speed up and slow down?
2. Will I receive fewer shares because the pool can’t decide my actual local hashrate?
3. Is this a bug?

Thank you.

Ps
My start.bat (it is run from within WalkingDeadMan.bat)
ethminer.exe -HWMON 1 -RH --v 5 --display-interval 30 --farm-recheck 2000 --cl-local-work 256 --cl-global-work 65536 --cl-parallel-hash 2 --cl-kernel 1 -G -P stratum1+tcp://0xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.xxxxxxxxxxxxx.rig1@eth-asia1.na..org@mymail

**Comment by AndreaLanfranchi** [comment]
> Does this have any bad effects on the GPUs as they constantly speed up and slow down?

Actually no. Hashrate fluctuation is only a matter about how we do compute avg hashrate on a period of 10 seconds. The higher the frequency your pool sends new jobs the higher the fluctuations you might see because for each new job GPUs stop hashing on the old one, set new parameters, and eventually begin to hash on the newer. This implies a time we call switch-time which is generally measured in a couple of tens of milliseconds.
The effective speed of hashing is the same.

> Will I receive fewer shares because the pool can’t decide my actual local hashrate?

Absolutely no. Pool does not care about your "reported" hashrate. You could even decide not to report hashrate at all. You want to report your hashrate as a term of comparison with the **effective** hashrate computed by pool which is derived from an _inverse_ math: given the known difficulty pool computes your effective hashrate based on the number of valid shares (stale or not) submitted in the unit of time.

> Is this a bug?

No.

**Comment by newtoopensuse** [comment]
Thanks. Still, I don’t get it. Suppose I have one gpu. If its hashrate is 30, it can process more jobs. But its hashrate is 28 now, it surely can only process fewer ones. So I think if a gpu can maintain a constant hashrate, it is more efficient and gets more shares. Am I correct?
Thank you.

**Comment by newtoopensuse** [comment]
I’m still confused. It takes 1 sec to finish the 5 jobs as you said. Then the gpu runs idle for 9 secs. Why are you saying the gpu worked for 9 secs?
And if I take your word for it, then why are people trying hard to increase their gpus’ speeds by modding and ramping up their memory clock? Isn’t it wiser to run the default bios and get the same effective hashrate with less heat and a cheaper energy bill?
You may be aware of Phoenix miner competing with claymore for a tiny increase in speed of 0.2 Mh.

Thank you.

PS.
In fact I modded my cards with srbpolaris 3.4 to mine Monero. Before this they did 820 each. After the mod they did 920 - a huge difference.

**Comment by AndreaLanfranchi** [comment]
> I’m still confused. It takes 1 sec to finish the 5 jobs as you said. Then the gpu runs idle for 9 secs. Why are you saying the gpu worked for 9 secs?

You're not following my words.
I did not say it takes 1 second to finish 5 jobs. 
I did say ... if, over a period of 10 seconds, pool sends 5 jobs, then ethminer has to switch job on GPU five times. Each "switching" operation has a duration, say, of 200ms. Thus (for 5 jobs) is a total of 1 sec. During this period of 1 second GPU is not actively hashing : is setting up.
So, again, during a period of 10 seconds, only 9 seconds were used to actively hash while 1 second was "wasted" in switch operations.

Overclocking then is a complete different beast.
First of all modding BIOS is not possible on NVIDIA: is available only for AMD.
Secondly when you overclock it's like you rev up your engine: you effectively get more speed.

Last word about Phoenix and Claymore. They're both closed source projects ... it's not easy to say wether or not they "fat" the reported performances for "marketing" reasons. Nevertheless, while using those miners, you pay a fee which, mostly, voids the better performance (if any).

**Comment by newtoopensuse** [comment]
Thank you very much. I understand why my gpus keep "jumping". But I think the devs can try to reduce the variance as claymore does. I mean you only see the digits to the right of the dot change (Sorry English is not my mother tongue) so the total speed seems to not change at all.

**Comment by MariusVanDerWijden** [comment]
We could display the average of one minute or longer, but how would that help? Only the displaying of the numbers would change not your effective hashrate. We don't want to bullshit our users by displaying hashrates that are unrealisticly smooth.

**Comment by newtoopensuse** [comment]
One more question. I see messages ‘the pool has sent a new job’. My understanding is 
1. if the gpu is fast enough, it doesn’t have to switch jobs and can submit more shares.
2. The pool doesn’t care about your reported hashrate, it just evenly distributes jobs to all workers and the one that is first to submit the finished job (share as I understand) wii be awarded ‘an Accepted’ share - that green word displayed on the console.

Thank you.

**Comment by AndreaLanfranchi** [comment]
> if the gpu is fast enough, it doesn’t have to switch jobs and can submit more shares.

Job switching is mandatory. When a pool sends in a new job it means the "old one" (the job currently your GPU is working on) is no more current. If you keep hashing on old job you will surely find new solutions but they will become firstly "stale" (for a few ms.) and then will be rejected as there is no work on pool's side which need shares for that work.

> The pool doesn’t care about your reported hashrate, it just evenly distributes jobs to all workers and the one that is first to submit the finished job (share as I understand) wii be awarded ‘an Accepted’ share - that green word displayed on the console.

Somehow yes.

**Comment by efrasyad** [comment]
>--cl-local-work 256 --cl-global-work 65536

from what one @jean-m-cyr told me, this will cause the switch time took longer as it will need to wait 65535 * 256 = 16MH compare to default (default --cl-global-work is 8192) 8192 * 256 = 2MH before the card start processing the actual new job, one of the reason high hashrate variants.
default --cl-global-work will give you lower hashrate variant (more stable hashrate)

**Comment by AndreaLanfranchi** [comment]
@efrasyad refer to this quite informative comment

https://github.com/ethereum-mining/ethminer/issues/977#issuecomment-378929052

**Comment by newtoopensuse** [comment]
Thank you all for your explanations and useful links.

**Comment by newtoopensuse** [comment]
Well, I followed the instructions on the page #977. I set --cl-global-work to 8192 and kept the rest unchanged. the total speed is now 169, which is about equal to claymore11.4's with -nofee 1. This is interesting as I read that claymore copied genoi code verbatim.
When I removed -nofee 1, the total speed went up to 176. Perhaps Andrea is right about claymore fatting the speed for marketing purposes. I'll let ethminer run and see if it will give the same profit as claymore.
